velox/functions/prestosql/window/tests/NtileTest.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/prestosql/window/tests/WindowTestBase.h"
+
+using namespace facebook::velox::exec::test;
+
+namespace facebook::velox::window::test {
+
+namespace {
+
+class NtileTest : public WindowTestBase {
```

Comment:
It is a bit hard to tell which code paths this test is testing. Does it test constant bucket? Does it test both null and non-null constant? Does it test cases where number of buckets >, =, < then number of partition rows? Does it test cases where buckets divide evenly or not over number of partition rows.

Reply:
These tests cover all those conditions. I've added a comment now for it.

There is an issue with testing null buckets pending as I see non-NULL results from DuckDB. That I'm investigating further. I will have a PR later on for it. Have left a comment here.